crates/kernel_cmdline/src/utf8.rs

```diff
@@ -213,6 +213,16 @@ impl<'a> Cmdline<'a> {
     }
 }
 
+impl Deref for Cmdline<'_> {
+    type Target = str;
+
+    fn deref(&self) -> &Self::Target {
+        // SAFETY: We know this is valid UTF-8 since we only
+        // construct the underlying `bytes` from valid UTF-8
+        str::from_utf8(&self.0).expect("We only construct the underlying bytes from valid UTF-8")
```

Comment:
`from_utf8` is a safe operation right? Isn't it `from_utf8_unchecked` that is the unsafe method?
Safe:
https://doc.rust-lang.org/std/primitive.str.html#method.from_utf8

> If you are sure that the byte slice is valid UTF-8, and you don’t want to incur the overhead of the validity check, there is an unsafe version of this function, [from_utf8_unchecked](https://doc.rust-lang.org/std/str/fn.from_utf8_unchecked.html), which has the same behavior but skips the check.

Unsafe:
https://doc.rust-lang.org/std/str/fn.from_utf8_unchecked.html

Just wondering if the safety comment is necessary here. Seems like it would only be necessary if using the `unchecked` version.